go/vt/vtctl/workflow/materializer.go

```diff
@@ -290,6 +290,15 @@ func (mz *materializer) deploySchema() error {
 		}
 	}
 
+	// Check if any table being moved is already non-empty in the target keyspace.
+	// Skip this check for multi-tenant migrations.
+	if !mz.IsMultiTenantMigration() {
+		err := validateEmptyTables(mz.ctx, mz.ts, mz.tmc, mz.targetShards, mz.ms.TableSettings)
+		if err != nil {
+			return err
```

Comment:
Nice IMO to wrap the error to add context:
```
return vterrors.Wrap(err, "failed to validate that all target tables are empty")
```

Reply:
done.